src/frontend/src/catalog/source_catalog.rs

```diff
@@ -121,6 +121,12 @@ impl SourceCatalog {
     ///
     /// Returns error if it's invalid.
     pub fn create_sql_ast_purified(&self) -> Result<ast::Statement> {
+        if self.with_properties.is_cdc_connector() {
+            // For CDC sources, we should not purify the SQL definition to add column definitions
+            // or constraints.
+            return self.create_sql_ast();
+        }
```

Comment:
I already forgot why/when to purify and not to purify. Would you mind adding some context and long term plan in the comment?